app/jobs/regular/chat_channel_delete.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Jobs
+  class ChatChannelDelete < ::Jobs::Base
+    def execute(args = {})
+      chat_channel = ::ChatChannel.with_deleted.find_by(id: args[:chat_channel_id])
+
+      # this should not really happen, but better to do this than throw an error
+      if chat_channel.blank?
+        Rails.logger.warn("Chat channel #{args[:chat_channel_id]} could not be found, aborting delete job.")
+        return
+      end
+
+      DistributedMutex.synchronize(
+        "delete_chat_channel_#{chat_channel.id}",
+        validity: 20.minutes
+      ) do
+        Rails.logger.debug("Deleting webhooks and events for channel #{chat_channel.id}")
+        ChatMessage.transaction do
+          webhooks = IncomingChatWebhook.where(chat_channel: chat_channel)
+          ChatWebhookEvent.where(incoming_chat_webhook_id: webhooks.pluck(:id)).delete_all
+          webhooks.delete_all
+        end
+
+        Rails.logger.debug("Deleting drafts and memberships for channel #{chat_channel.id}")
+        ChatDraft.where(chat_channel: chat_channel).delete_all
+        UserChatChannelMembership.where(chat_channel: chat_channel).delete_all
+
+        Rails.logger.debug("Deleting chat messages, mentions, revisions, and uploads for channel #{chat_channel.id}")
+        ChatMessage.transaction do
+          chat_messages = ChatMessage.where(chat_channel: chat_channel)
```

Comment:
A minor performance concern here, potentially how many chat messages could we be dealing with? Perhaps it might make sense at some point to break this into smaller jobs that can execute in parallel?

Reply:
Hmm yeah I did think about that briefly, could be thousands I suppose :/ I could just do find_in_batch like I do for archiving:

```
      chat_channel.chat_messages.find_in_batches(
        batch_size: ARCHIVED_MESSAGES_PER_POST
      ) do |chat_messages|
```

And delete batches of them at a time

Reply:
This wouldn't be a problem now that we're not plucking the `id`s. Previously, each pluck will allocate an array the size of the number of messages in the channel.